webapp/src/shared/components/Button/Button.tsx

```diff
@@ -0,0 +1,43 @@
+import { ReactNode } from 'react';
+import './Button.css';
+
+export enum ButtonTypes {
+  SUBMIT = 'submit',
+  WARNING = 'warning',
+}
+
+type Buttonprops = {
+  buttonType: ButtonTypes;
+  buttonIcon?: JSX.Element;
```

Comment:
Sorry, but I'm not able to do it like that.
I get this error message:
```
type Icon = /*unresolved*/ any
'Icon' refers to a value, but is being used as a type here. Did you mean 'typeof Icon'?
```
And if use `typeof Icon` I get other errors

Reply:
One option is to pass the `IconVariant` as a prop instead of an `Icon`.

Something like this:
<details>

```diff
diff --git a/webapp/src/pages/Landing/Landing.tsx b/webapp/src/pages/Landing/Landing.tsx
index a6bb669..f8444d1 100644
--- a/webapp/src/pages/Landing/Landing.tsx
+++ b/webapp/src/pages/Landing/Landing.tsx
@@ -13,13 +13,6 @@ import { Color } from '../../shared/types';
 import './Landing.css';
 
 export default function Landing() {
-  const btn_icon = (
-    <Icon
-      variant={IconVariant.LOGOUT}
-      color={Color.LIGHT}
-      size={IconSize.SMALL}
-    ></Icon>
-  );
   const btn_action = (): void => alert('This is an alert');
   const btn_link = (): void => {
     window.location.href = 'http://google.com';
@@ -29,18 +22,16 @@ export default function Landing() {
       <Button variant={ButtonVariant.SUBMIT} onClick={btn_action}>
         ALERT
       </Button>
-      <Button variant={ButtonVariant.WARNING} icon={btn_icon} disabled>
+      <Button
+        variant={ButtonVariant.WARNING}
+        iconVariant={IconVariant.LOGOUT}
+        disabled
+      >
         LOGOUT BUTTON
       </Button>
       <Button
         variant={ButtonVariant.SUBMIT}
-        icon={
-          <Icon
-            variant={IconVariant.ARROW_FORWARD}
-            color={Color.DARK}
-            size={IconSize.SMALL}
-          ></Icon>
-        }
+        iconVariant={IconVariant.ARROW_FORWARD}
         onClick={btn_link}
       >
         To Google
@@ -49,7 +40,7 @@ export default function Landing() {
         variant={IconVariant.PLAY}
         color={Color.ONLINE}
         size={IconSize.LARGE}
-      ></Icon>
+      />
       <Text
         variant={TextVariant.TITLE}
         color={TextColor.GAME}
diff --git a/webapp/src/shared/components/Button/Button.tsx b/webapp/src/shared/components/Button/Button.tsx
index 5f813ae..77c3523 100644
--- a/webapp/src/shared/components/Button/Button.tsx
+++ b/webapp/src/shared/components/Button/Button.tsx
@@ -1,3 +1,4 @@
+import Icon, { IconSize, IconVariant } from '../Icon/Icon';
 import './Button.css';
 
 export enum ButtonVariant {
@@ -7,7 +8,7 @@ export enum ButtonVariant {
 
 type ButtonProps = {
   variant: ButtonVariant;
-  icon?: JSX.Element;
+  iconVariant?: IconVariant;
   onClick?: () => void;
   disabled?: boolean;
   children: string;
@@ -15,7 +16,7 @@ type ButtonProps = {
 
 export default function Button({
   variant,
-  icon,
+  iconVariant,
   onClick,
   disabled = false,
   children,
@@ -26,7 +27,11 @@ export default function Button({
       disabled={disabled}
       onClick={onClick}
     >
-      {icon && <div className="button-icon">{icon}</div>}
+      {iconVariant && (
+        <div className="button-icon">
+          <Icon variant={iconVariant} size={IconSize.SMALL} />
+        </div>
+      )}
       {children}
     </button>
   );

```

</details>

I think this will also couple the `Icon` color with the color of the `Button` text.

Reply:
Yes, that's a good option. Let's see what @vdedios says

Reply:
Yes, totally agree with that option :+1: